docs/source/tutorials/ax-blop-demo.ipynb

```diff
@@ -0,0 +1,331 @@
+    "\n",
+    "Ax expects a `tuple[float, float]` for each objective representing the mean value and standard error, respectively. For a single image, the average intensity is just the intensity (same for the area), and we assume no uncertainty."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "from blop.utils import get_beam_stats\n",
+    "\n",
+    "\n",
+    "def evaluate(results_df: pd.DataFrame) -> dict[str, tuple[float, float]]:\n",
+    "    stats = get_beam_stats(results_df[\"bl_det_image\"].iloc[0])\n",
+    "    area = stats[\"wid_x\"] * stats[\"wid_y\"]\n",
+    "    return {\n",
+    "        \"beam_intensity\": (stats[\"sum\"], None),\n",
+    "        \"beam_area\": (area, None),\n",
+    "    }"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Create an evaluator\n",
+    "\n",
+    "We need a Bluesky evaluator that actually launches the experiment using the `RunEngine` and retreives the result using `Databroker`. Here we need to specify which detectors will produce the image as well as which motors we will be moving. Also, we pass the evaulation function here to produce the objective values.\n",
+    "\n",
+    "This evaluator will be used to produce the raw data needed by Ax to optimize the parameters to satisfy our objectives."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from blop.integrations.ax import create_bluesky_evaluator\n",
+    "\n",
+    "\n",
+    "evaluator = create_bluesky_evaluator(\n",
+    "    RE, db, [beamline.det], [beamline.kbv_dsv, beamline.kbv_usv, beamline.kbh_dsh, beamline.kbh_ush], evaluate\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Optimize!\n",
+    "\n",
+    "Finally, with all of our experimental setup done, we can optimize the parameters to satisfy our objectives.\n",
+    "\n",
+    "For this example, Ax will optimize the 4 motor positions to produce the greatest intensity beam with the smallest beam width and height (smallest area). It does this by first running a couple of `Trial`s which are random samples, then the remainder using Bayesian optimization through BoTorch.\n",
+    "\n",
+    "A single Ax `Trial` represents the training and evaluation of BoTorch models to produce a suggested next `Arm`. An `Arm` in Ax is a single parameterization to be evaluated while a `Trial` can consist of many `Arm`s. In this demo, we have a single `Arm` per `Trial`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "for _ in range(25):\n",
```

Comment:
Question about Ax, you picked 0-25 as the range for optimization, does more trials improve the optimization process or is it just like an ensemble and averages out at the end? 
I see in the markdown you mention the first few are random and rest is BO.

Reply:
Ax actually infers what optimization setup to run based on the parameters you pass in according to the docs: https://ax.readthedocs.io/en/stable/service.html#module-ax.service.ax_client.

If you want something more custom then you can pass in your own `GenerationStrategy` (which suggests new parameters, or `Arm`s, to try).